### PR TITLE
Various commits to reduce parallelism in LocalPublisherManager

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-13410
https://issues.redhat.com/browse/ISPN-13397

This PR includes a few different issues with having too much parallelism within the LocalPublisherManager. These changes ensure that the parallelism is limited to the number of CPUs and doesn't scale higher than that.

### CODE REVIEW

File: core/src/test/java/org/infinispan/reactive/publisher/impl/SimpleLocalPublisherMangerTest.java [hunk 1]
```diff
@@ -20,28 +24,34 @@
...
 import org.testng.annotations.Test;
 
 import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.functions.BiFunction;
 
 @Test(groups = "functional", testName = "reactive.publisher.impl.SimpleLocalPublisherMangerTest")
 @InCacheMode({CacheMode.REPL_SYNC, CacheMode.DIST_SYNC, CacheMode.SCATTERED_SYNC})
 public class SimpleLocalPublisherMangerTest extends MultipleCacheManagersTest {
+   private static final int SEGMENT_COUNT = 128;
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/reactive/publisher/impl/SimpleLocalPublisherMangerTest.java:43]:
```suggestion
public class SimpleLocalPublisherManagerTest extends MultipleCacheManagersTest {
   private static final int SEGMENT_COUNT = 128;
```

File: core/src/test/java/org/infinispan/reactive/publisher/impl/SimpleLocalPublisherMangerTest.java [hunk 2]
```diff
@@ -94,4 +104,65 @@ public void testNoIntermediateOps(DeliveryGuarantee deliveryGuarantee, boolean i
...
+
+      io.reactivex.rxjava3.functions.Function<Object, Single<Object>> sleepOnBlockingPoolFunction = value ->
+            Single.fromCompletionStage(blockingManager.supplyBlocking(() -> value, "test-blocking-thread"));
+
+      if (isEntry) {
+         stage = lpm.keyReduction(isParallel, allSegments, null, null, false,
+               deliveryGuarantee, publisher -> Flowable.fromPublisher(publisher)
+                     .concatMapSingle(sleepOnBlockingPoolFunction).collect(collector).toCompletionStage()
+               , publisher -> Flowable.fromPublisher(publisher)
+                     .reduce(reduceBiFunction).toCompletionStage(Collections.emptySet()));
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/reactive/publisher/impl/SimpleLocalPublisherMangerTest.java:143]: Nitpicking maybe, but the orphan `,` really bugs me. I like the one line per operation format much better, although in the suggestion box it looks worse.

```suggestion
         stage = lpm.keyReduction(isParallel, allSegments, null, null, false, deliveryGuarantee,
                                  publisher -> Flowable.fromPublisher(publisher)
                                                       .concatMapSingle(sleepOnBlockingPoolFunction)
                                                       .collect(collector)
                                                       .toCompletionStage(),
                                  publisher -> Flowable.fromPublisher(publisher)
                                                       .reduce(reduceBiFunction)
                                                       .toCompletionStage(Collections.emptySet()));
```

**Comment by danberindei** [comment]
Looks good, just 2 tiny requests

File: core/src/test/java/org/infinispan/reactive/publisher/impl/SimpleLocalPublisherMangerTest.java (hunk 2 above)
Comment (wburns) [reply core/src/test/java/org/infinispan/reactive/publisher/impl/SimpleLocalPublisherMangerTest.java:143]: Sure, I will see how it comes out.

**Comment by wburns** [comment]
Updated. I just squashed the latest two changes into the prior commit.

**Review by danberindei — APPROVED** [review]

**Comment by wburns** [comment]
Fixed test name failure, hopefully CI is better now.

**Comment by danberindei** [comment]
Merged, thanks Will!

**Pull request merged**